commit 0cf7359c075f8e0590f4f95aac789373658ae7e7
Author: Matthew Garrett <matthew.garrett@nebula.com>
Date:   Tue Nov 22 08:46:17 2016 +0000

    UBUNTU: SAUCE: (efi-lockdown) x86: Restrict MSR access when the kernel is locked down
    
    Writing to MSRs should not be allowed if the kernel is locked down, since
    it could lead to execution of arbitrary code in kernel mode.  Based on a
    patch by Kees Cook.
    
    Cc: Kees Cook <keescook@chromium.org>
    Signed-off-by: Matthew Garrett <matthew.garrett@nebula.com>
    Signed-off-by: David Howells <dhowells@redhat.com>
    (cherry picked from commit 2fb72aaa251fd38d09ae6da9c64cd129349b9812
     git://git.kernel.org/pub/scm/linux/kernel/git/jwboyer/fedora.git)
    Signed-off-by: Seth Forshee <seth.forshee@canonical.com>

diff --git a/arch/x86/kernel/msr.c b/arch/x86/kernel/msr.c
index ef688804f80d..fbcce028e502 100644
--- a/arch/x86/kernel/msr.c
+++ b/arch/x86/kernel/msr.c
@@ -84,6 +84,9 @@ static ssize_t msr_write(struct file *file, const char __user *buf,
 	int err = 0;
 	ssize_t bytes = 0;
 
+	if (kernel_is_locked_down())
+		return -EPERM;
+
 	if (count % 8)
 		return -EINVAL;	/* Invalid chunk size */
 
@@ -131,6 +134,10 @@ static long msr_ioctl(struct file *file, unsigned int ioc, unsigned long arg)
 			err = -EBADF;
 			break;
 		}
+		if (kernel_is_locked_down()) {
+			err = -EPERM;
+			break;
+		}
 		if (copy_from_user(&regs, uregs, sizeof regs)) {
 			err = -EFAULT;
 			break;
